trin-core/src/portalnet/overlay_service.rs

```diff
@@ -14,12 +16,19 @@ use crate::{
         },
         Enr,
     },
-    utils::distance::xor_two_values,
+    utils::{distance::xor_two_values, hash_delay_queue::HashDelayQueue},
 };
 
-use discv5::{enr::NodeId, kbucket::KBucketsTable};
-use futures::channel::oneshot;
-use log::{debug, info};
+use discv5::{
+    enr::NodeId,
+    kbucket::{
+        self, ConnectionDirection, ConnectionState, FailureReason, InsertResult, KBucketsTable,
+        NodeStatus, UpdateResult,
+    },
+    rpc::RequestId,
+};
+use futures::{channel::oneshot, stream::StreamExt};
```

Comment:
For `oneshot`, I don't have a reason to prefer one crate over the other, so I'm happy to move it to `tokio::sync::oneshot` if someone prefers that.

We need to use `StreamExt` so that we can call `next` on the ping queue.